### PR TITLE
Feat/typescript generator

### PR DESCRIPTION
**Pull request by xiboXD**

### CODE REVIEW

**Review by eanzhao — APPROVED** [review]

File: aelf.tools/AElf.Tools/build/_typescript/node_modules/.bin/protoc
Comment (aelf-lu) [inline aelf.tools/AElf.Tools/build/_typescript/node_modules/.bin/protoc:1]: Can I compress node_modules folder into zip package? Unzip it before using it in msbuild

File: aelf.tools/AElf.Tools/AElf.Tools.csproj
Comment (aelf-lu) [inline aelf.tools/AElf.Tools/AElf.Tools.csproj:1]: NoDefaultExcludes and NoWarn should be moved to an new <PropertyGroup>

Comment (xibo1) [reply aelf.tools/AElf.Tools/AElf.Tools.csproj:1]: done

Comment (xiboXD) [reply aelf.tools/AElf.Tools/AElf.Tools.csproj:1]: done

**Pull request closed without merging**